great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -4787,21 +4783,6 @@ def include_rendered_content(self) -> IncludeRenderedContentConfig:
     def notebooks(self) -> NotebookConfig:
         return self.variables.notebooks  # type: ignore[return-value]
 
-    @property
-    def datasources(
```

Comment:
I didn't think we'd get rid of this property, but instead, have it fetch the full list of datasources.

There are definitely other places in the code that depend on this property. Some of them are incorrect as they seem to mutate it, meaning that the config won't be persisted